### PR TITLE
Adjusting 1x1 HF output LUTs

### PR DESCRIPTION
**Pull request by akhukhun**
Make 1x1 HF output LUTs 4:1 to keep the TP lsb same as for 3x2 (0.5 GeV).

### CODE REVIEW

**Comment by mmusich** [comment]
please test

**Comment by davidlange6** [comment]
@akhukhun - could you summarize in more detail the change?

**Comment by akhukhun** [comment]
For 3x2 TPs there should be no change. 
For 1x1 TPs, the LSB  before the change is 1/8 GeV. 
The commit changes the output (compression) LUTs from identity (1:1) to 4:1, resulting in 1x1 HF TP LSB = 0.5 GeV -- the same as for 3x2 TPs.

**Comment by mmusich** [comment]
@akhukhun does this imply changes as well in the conditions?

**Comment by akhukhun** [comment]
No change in the offline conditions will be needed. We will have the corresponding output LUTs added in the online LUTs.

**Comment by mmusich** [comment]
+1

**Comment by abdoulline** [comment]
From HCAL internal discussion: need it in both in 80X( this one) and 81X(will be submitted soon) to correctly simulate TPs...

**Comment by fwyzard** [comment]
@akhukhun how does one configure the L1 to pick up the corresponding LUTs ?

**Comment by akhukhun** [comment]
As far as I know Layer-1 has already switched to using  CaloTPGTranscoderULUT class to convert HCAL TPs to GeV, (as we have agreed), so the scale should be correct automatically. With this commit merged, the TPs won't saturate at 32 GeV -- that's the difference the commit should make in the end. 

P.S. if one uses the Layer-1 version where it does not use CaloTPGTranscoderULUT but assumes the HF LSB of 0.5 GeV, the scale after this PR will still be correct, but off by factor of 4 before this PR.

**Comment by rekovic** [comment]
+1

L1T already looked at the effect of new HF TP LSBs (https://github.com/cms-l1t-offline/cmssw/issues/202).  
Expect a small increase in rate (caviat: conclusion not derived on ZeroBias but on 2015 MuonEG data.)  Layer-1 experts also looked at the effect, and confirmed that 32 GeV threshold in TPs is gone.

**Comment by fwyzard** [comment]
@Martin-Grunewald , @perrotta , pending the next 8.0.x release, could you add this PR to the TSG instructions ?

**Comment by Martin-Grunewald** [comment]
@fwyzard 
Done!

**Comment by kpedro88** [comment]
@davidlange6 - reminder...

**Comment by davidlange6** [comment]
keep in mind the equiv.  PR has been crashing workflows the IBs in 81x - the fix came ~yesterday

**Comment by kpedro88** [comment]
Ah, didn't realize that. @akhukhun, once the fix is verified, please make sure to port it back to this PR.

File: CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.h [hunk 1]
```diff
@@ -65,6 +65,7 @@ class CaloTPGTranscoderULUT : public CaloTPGTranscoder {
   std::vector<int> ZS;
   std::vector<int> LUTfactor;
 
+  unsigned int size;
```
Comment (fwyzard) [inline CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.h]: this variable seems to be unused ?

Comment (fwyzard) [inline CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.h]: Ah, no, my mistake, found it.

**Comment by fwyzard** [comment]
please test

**Comment by mmusich** [comment]
+1

**Comment by abdoulline** [comment]
@mulhearn   
we need it for correct HCAL TPs emulation in 80X

**Comment by rekovic** [comment]
+1

**Comment by davidlange6** [comment]
+1
